### DISCUSSION TITLE
Macports exiftool - fix.

### DISCUSSION THREAD

**Discussion by jonwilts** [RhetTbull/osxphotos#1733] [General]
Hello. I'm a Macports user, and so the python, perl, exiftool, etc. installs I use come from Macports.

Recently (I'm a sporadic user so can't say exactly when) osxphotos stopped working for me - it would appear to hang at the point it tried to actually export a photo.

After some debugging and pointers from the r/osxphotos subreddit, I discovered the problem was caused by an upstream change to exiftool, which changed the 1st line of the exiftool script to `#!/usr/bin/env perl -w`. Unfortunately, when exiftool was run by osxphotos, this new first line returned system perl, not Macports perl, which in turn meant that the `Image::ExifTool` library wasn't in perl's @inc path. Annoyingly, `exiftool` worked from the command line, but `osxphotos --exiftool` didn't.

The fix was to edit exiftool to use `#!/opt/local/bin/perl -w` and now everything works again.

Posting this in case anyone else runs into this issue; have a couple of hours of your life back!

**Comment by RhetTbull** [comment]
Glad you figured it out. Thanks for posting the fix!
